src/lib/components/quiz/quiz.svelte

```diff
@@ -0,0 +1,233 @@
+  import RightArrow from '../icons/rightArrow.svelte';
+  import MoreInfo from './moreInfo.svelte';
+
+  export let data: QuizData;
+
+  $: index = 0;
+  let response: string[] = [];
+  let noMoreQuestions: boolean = false;
+  let answeredEveryQuestion: boolean = false;
+  let showResults: boolean = false;
+  let showMoreInfo: boolean = false;
+  let showTeam: string;
+  let tallyChoices = {
+    aiChoice: 0,
+    algoChoice: 0,
+    designChoice: 0,
+    devChoice: 0,
+  };
+  let teams = ['aiChoice', 'algoChoice', 'designChoice', 'devChoice'];
+  let match: string;
+
+  function goLeft() {
+    if (index > 0) {
+      index--;
+      noMoreQuestions = false;
+    }
+  }
+
+  function goRight() {
+    if (index < data.questions.length - 1) {
+      index++;
+    }
+    if (index === data.questions.length - 1) {
+      noMoreQuestions = true;
+
+      console.log('please answer all questions');
+    }
+  }
+
+  function recordAnswer(choiceIndex: number) {
+    response[index] = data.questions[index].choices[choiceIndex].match;
+    // if answered all questions and no more questions then submitResponses
+    if (!response.includes(undefined) && response.length === data.questions.length) {
+      answeredEveryQuestion = true;
+    } else {
+      goRight();
+    }
+  }
+
+  function submitResponses() {
+    // tally up response choices once it is locked in and the user cant go back
+    // below will condense the total number of clicks for its respective button
+    // ex: {ai: 1, algo: 2, dev: 0, design: 0}
+    response.forEach((choice) => {
+      tallyChoices[choice] += (1 / data.questions.length) * 100;
+    });
+    // calculate best match
+    let totalTallyValues = Object.values(tallyChoices);
+    let totalTallyKey = Object.keys(tallyChoices);
+    // first we find the biggest number => get index of that => find corresponding key to that index
+    match = totalTallyKey[totalTallyValues.indexOf(Math.max(...totalTallyValues))];
+    // save to localstorage
+    // show results and hide the question
+    showResults = true;
+  }
+
+  function restartQuiz() {
+    response = [];
+    index = 0;
+    showResults = false;
+    noMoreQuestions = false;
+    answeredEveryQuestion = false;
+    tallyChoices = {
+      aiChoice: 0,
+      algoChoice: 0,
+      designChoice: 0,
+      devChoice: 0,
+    };
+  }
+
+  function showTeamDetails(currentTeam: string) {
+    showMoreInfo = true;
+    showTeam = currentTeam;
+  }
+  function goBackToResults() {
+    showMoreInfo = false;
+  }
+</script>
+
+<div class="container">
+  <!-- DISPLAY THE QUIZ QUESTIONS -->
+  {#if !showResults}
+    <h1>ACM PATH QUIZ</h1>
+    <div class="question">
+      <h2>{data.questions[index].prompt}</h2>
+      {#if response[index]}
+        <p>answered</p>
+      {/if}
+      <section class="answers">
+        {#each data.questions[index].choices as choice, j (choice.content)}
+          <button on:click={() => recordAnswer(j)} style={`--color: ${choice.color}`}>
```

Comment:
We could pass `match` to the `recordAnswer` function directly.
```ts
  function recordAnswer(match: TeamMatch) {
```
```suggestion
        {#each data.questions[index].choices as choice (choice.content)}
          <button on:click={() => recordAnswer(choice.match)} style={`--color: ${choice.color}`}>
```

Reply:
Done